File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    let answer = '';
    
    const memo = {};
    
    participant.forEach((name, index) => {
        if (memo[name]) {
            ++memo[name];
        } else {
            memo[name] = 1;
        }
    });
    
    completion.forEach(name => {
        if (--memo[name] === 0) {
            delete memo[name];
        }
    });
    
    answer = Object.keys(memo)[0];
    
    return answer;
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    
    map<string, int> completionNames;
    
    for (string person : completion)
    {
        if (completionNames[person] == 0)
            completionNames[person] = 1;
        else
            ++completionNames[person];
    }
    
    for (string person : participant)
    {
        if (completionNames[person] > 0)
            --completionNames[person];
        else
        {
            answer = person;
            break;
        }
    }
    
    
    return answer;
}